framework/src/vectorpostprocessors/ElementVariablesDifferenceMax.C

```diff
@@ -0,0 +1,126 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "ElementVariablesDifferenceMax.h"
+
+// MOOSE includes
+#include "MooseVariable.h"
+
+// libMesh includes
+#include "libmesh/quadrature.h"
+
+template <>
+InputParameters
+validParams<ElementVariablesDifferenceMax>()
+{
+  InputParameters params = validParams<ElementVectorPostprocessor>();
+
+  params.addRequiredCoupledVar(
+      "compare_a",
+      "The first variable to evaluate the difference with, performed as \"compare_a - compare_b\"");
+  params.addRequiredCoupledVar("compare_b",
+                               "The second variable to evaluate the difference with, "
+                               "performed as \"compare_a - compare_b\"");
+
+  params.addParam<bool>(
+      "furthest_from_zero", false, "Find the difference with the highest absolute value");
+
+  return params;
+}
+
+enum CollectionOfAllValuesIntoAVector
```

Comment:
`CollectionOfAllValuesIntoAVector` is for easy indexing of the vector `_all`. If I use a strong-typed `enum class` then this feature is no longer implicitly available. Instead, it would require a cast each time I need to use a value.

The original intent was to provided readability and maintainability to the code.